controllers/substitution.go

```diff
@@ -0,0 +1,44 @@
+package controllers
+
+import (
+	"strings"
+
+	v1 "k8s.io/api/apps/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	// commonControllerNamingSubstringPattern contains a substring which all CCM controller container names should have
+	// to be substituted with image from config
+	commonControllerNamingSubstringPattern = "controller"
```

Comment:
I think the pattern here is good, might be better to use something explicit though, eg, if we have a list of predefined allowed values `cloud-controller-manager`, `node-controller-manager` etc that the manifests must obide by? WDYT?

Reply:
We need some sort of CI task which will check that only those names are on containers and other subresources all the time.